quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -560,71 +585,157 @@ impl IngestController {
         shard_stats: ShardStats,
         model: &mut ControlPlaneModel,
         progress: &Progress,
-    ) {
+    ) -> MetastoreResult<()> {
         const NUM_PERMITS: u64 = 1;
 
         if !model
             .acquire_scaling_permits(&source_uid, ScalingMode::Up, NUM_PERMITS)
             .unwrap_or(false)
         {
-            return;
+            return Ok(());
         }
-        let new_num_open_shards = shard_stats.num_open_shards + 1;
 
-        info!(
-            index_id=%source_uid.index_uid.index_id,
-            source_id=%source_uid.source_id,
-            "scaling up number of shards to {new_num_open_shards}"
-        );
-        let unavailable_leaders: FnvHashSet<NodeId> = FnvHashSet::default();
+        let new_num_open_shards = shard_stats.num_open_shards + 1;
 
-        let Some((leader_id, follower_id)) = self
-            .allocate_shards(1, &unavailable_leaders, model)
-            .and_then(|pairs| pairs.into_iter().next())
-        else {
-            warn!("failed to scale up number of shards: no ingesters available");
-            model.release_scaling_permits(&source_uid, ScalingMode::Up, NUM_PERMITS);
-            return;
-        };
-        let shard_id = ShardId::from(Ulid::new());
-        let open_shard_subrequest = metastore::OpenShardSubrequest {
-            subrequest_id: 0,
-            index_uid: source_uid.index_uid.clone().into(),
-            source_id: source_uid.source_id.clone(),
-            shard_id: Some(shard_id),
-            leader_id: leader_id.into(),
-            follower_id: follower_id.map(Into::into),
-        };
-        let open_shards_request = metastore::OpenShardsRequest {
-            subrequests: vec![open_shard_subrequest],
-        };
-        let open_shards_response = match progress
-            .protect_future(self.metastore.open_shards(open_shards_request))
-            .await
-        {
-            Ok(open_shards_response) => open_shards_response,
-            Err(error) => {
-                warn!("failed to scale up number of shards: {error}");
+        let new_shard_source_uids: HashMap<SourceUid, usize> =
+            std::iter::once((source_uid.clone(), 1)).collect();
+        let successful_source_uids_res = self
+            .try_open_shards(new_shard_source_uids, model, &Default::default(), progress)
+            .await;
+        match successful_source_uids_res {
+            Ok(successful_source_uids) => {
+                assert!(successful_source_uids.len() <= 1);
+                if successful_source_uids.is_empty() {
+                    // We did not manage to create the shard.
+                    // We can release our permit.
+                    model.release_scaling_permits(&source_uid, ScalingMode::Up, NUM_PERMITS);
+                    warn!(
+                        index_id=%source_uid.index_uid.index_id,
```

Comment:
```suggestion
                        index_uid=%source_uid.index_uid,
```
I think logging the `index_uid` is preferable whenever possible.